src/lib/pickles/verify.ml

```diff
@@ -148,31 +148,49 @@ let verify_heterogenous (ts : Instance.t list) =
           in
           (absorb sponge, squeeze)
         in
```

Comment:
btw I'm trying to understand what this function does. It's called `verify_heterogenous`, so I assume it's shared between both circuits, but if this is the case why does it uses Tick only?

Reply:
It's called `verify_heterogeneous` because we verify proofs with multiple different verification keys  etc. all at once. We're always verifying a wrap proof here.